app/models/course/condition/achievement.rb

```diff
@@ -3,4 +3,47 @@ class Course::Condition::Achievement < ActiveRecord::Base
   belongs_to :achievement, class_name: Course::Achievement.name
 
   default_scope { includes(:achievement) }
+
+  validate :valid_achievement_condition, if: :achievement_id_changed?
```

Comment:
it's not about the _id... the thing is, the _id might be null when the non-id attribute might have the value defined.

in those cases, the object associated hasn't been saved. how does your validation cope with such a case, if it is necessary?